File: validation/merge_qa/verify_appendix.ts
import fs from "node:fs/promises";
import { DocxPackage } from "../../server/src/docx/package.js";
import { buildDocumentModel } from "../../server/src/docx/model.js";
import { countWords } from "../../server/src/merge/word_count.js";

async function main() {
  const buffer = await fs.readFile("validation/merge_qa/merged-9000-v3.docx");
  const model = await buildDocumentModel(await DocxPackage.load(buffer));
  const ignore = model.paragraphs.findIndex((paragraph) => paragraph.text.trim() === "IGNORE");
  const appendixWords = model.paragraphs.slice(ignore + 1).reduce((sum, paragraph) => sum + countWords(paragraph.text), 0);
  console.log({ appendixWords, ignore, tables: model.tables.length, images: model.imageCount });
}

void main();

File: validation/merge_qa/verify_appendix.py
import re
from docx import Document

document = Document("validation/merge_qa/merged-9000-v2.docx")
paragraphs = [paragraph.text for paragraph in document.paragraphs]
ignore_index = paragraphs.index("IGNORE")
pattern = re.compile(r"[\w]+(?:['’.-][\w]+)*", re.UNICODE)
appendix_text = " ".join(paragraphs[ignore_index + 1:])
all_text = " ".join(paragraphs)
print("appendix_words", len(pattern.findall(appendix_text)))
print("ignore", paragraphs.count("IGNORE"))
print("removed_references", all_text.count("Removed Author"), all_text.count("Delete Me"))
print("tables", len(document.tables), "images", len(document.inline_shapes), "paragraphs", len(paragraphs))
